mlx_lm/models/rope_utils.py

```diff
@@ -65,7 +65,13 @@ def default_scale(factor):
 
     def __call__(self, x, offset: int = 0):
         seq_len = offset + x.shape[-2]
-        if seq_len > self.original_max_position_embeddings:
+
+        if isinstance(seq_len, mx.array):
+            seq_len_scalar = seq_len.max().item()
+        else:
```

Comment:
We shouldn't do a graph evaluation in each call to rope. This will incur quite a bit of latency and potentially break async evaluation.

I also don't love the alternative of doing a `mx.where(seq_len.max() > self.original..., self._long_freq, self._short_freqs)`.